design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,169 @@
+# Design Proposal: Project Deployment with Kubernetes Operator
+
+**Author**: @kfirtoledo
+
+**Begin Design Discussion**: 2023-11-30
+
+**Status:** draft
+
+## Summary/Abstract
+
+The ClusterLink project allows the interconnection of various clusters within the same fabric. For each fabric, there is a fabric administrator responsible for determining which clusters can join the fabric, and a site administrator responsible for deploying ClusterLink to the Kubernetes (K8s) cluster.
+
+This design proposal outlines the deployment process for the ClusterLink project on a Kubernetes cluster by the site administrator. It relies on a dedicated K8s operator to make the deployment more user-friendly, simplified, and native to K8s users.
+
+## Background
+
+### Motivation and problem space
+
+This design proposal aims to simplify and enhance the deployment of the ClusterLink project.
+By leveraging K8s native mechanisms, such as operators and CRDs, the objective is to establish a user-friendly deployment process.
+Currently, ClusterLink deployment involves using the cl-adm CLI, which generates a large YAML file that contains the control plane and dataplane deployments, services, and RBAC configurations.
+In the new design, this process will be managed by the ClusterLink operator, which has the following advantages:
+
+* It consumes a simpler and more concise YAML configuration.
+* The use of CRD and the operator aids in automating deployment changes over time.
+* The proposal differentiates between fabric and site administrators and provides a more secure deployment model via k8s RBAC.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve and simplify the deployment process for the ClusterLink users.
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+As a site administrator, I would like to join my cluster to an existing fabric. The fabric administrator should send me the certificates required to join the fabric: the fabric CA certificate and the cluster certificates (private and public keys). Using these certificates and the ClusterLink operator, I can deploy the ClusterLink system in my cluster. Now, any user in my cluster can connect to other remote services in the fabric, even if they are located on different locations, or different cloud providers.
```

Comment:
Another potential story is deploy / undeploy of ClusterLink. It somewhat overlaps with the above, but is more focused on what is being addressed by this design (IMHO, the "joining a fabric" is more about the exchange of certificates and then the deployment - but maybe that's just my world view...)